test/controllers/api/v2/content_views_controller_test.rb

```diff
@@ -170,6 +171,21 @@ def test_update_components
       assert_template 'katello/api/v2/common/update'
     end
 
+    def test_duplicate_component_error_message
+      view = create(:katello_content_view)
+      view_version = create(:katello_content_view_version, :content_view => view)
+
+      composite = create(:katello_content_view, :composite)
```

Comment:
Now that you have a solid test case I think it would make sense to use existing fixtures wherever possible. For this test I think that means we can use existing fixtures for both content views but you can continue to use FactoryBot for creating the necessary versions. The main idea is to avoid running SQL when possible and we can do that here by avoiding creating new objects and instead reuse existing ones